pallets/evercity/src/bond.rs

```diff
@@ -130,60 +130,52 @@ pub struct BondInnerStruct<Moment, Hash> {
     pub impact_data_type: BondImpactType,
     /// Base value Now, all types has same interest_rate calculation logic
     /// greater the value -> lower the interest_rate and vice-versa
-    pub impact_data_baseline: Vec<u64>,
+    pub impact_data_baseline: Vec<Option<u64>>,
```

Comment:
Why not `Option<Vec<u64>>`?

Reply:
It was failing later in the code when getting indices from `impact_data_baseline`

Reply:
I guess if `bond.inner.is_stable() == false` the `impact_data_baseline` is (must be) guaranteed to be `Some`, so it should be safe to just `unwrap()` it

Reply:
Makes sense, will try

Reply:
No, it  fails with all the length and indices tests
`error[E0608]: cannot index into a value of type `std::option::Option<std::vec::Vec<u64>>`
    --> pallets/evercity/src/lib.rs:1891:25
     |
1891 |                         bond.inner.impact_data_baseline[index - 1].unwrap_or(0),
     |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `len` found for enum `std::option::Option<std::vec::Vec<u64>>` in the current scope
   --> pallets/evercity/src/bond.rs:284:46
    |
284 |                 && self.impact_data_baseline.len() == self.bond_duration as usize
    |                                              ^^^ method not found in `std::option::Option<std::vec::Vec<u64>>``

Reply:
Right, here you need add a check that `self.impact_data_baseline.is_some()` and then you can safely unwrap it.